### PR TITLE
Specs

### PR DESCRIPTION
**Pull request by irubnich**
Asana: https://app.asana.com/0/8174318543888/19679344516444

Todo:
- [x] Fix the random spec failure
- [x] Regex test name matching?
- [x] Specs for specs
- [x] Update README

### CODE REVIEW

File: bin/kontrast [hunk 1]
```diff
@@ -77,7 +78,7 @@ module Kontrast
         end
 
         private
-            def load_config(config)
+            def load_config(config, specs_path = nil)
```
Comment (werkshy) [inline bin/kontrast]: Per our code conventions this should be a named arg, and it probably should be here too.

File: lib/kontrast/runner.rb [hunk 2]
```diff
@@ -20,95 +17,89 @@ def run
                 current_node = 0
             end
 
+            # Load & bind specs
+            Kontrast.test_suite.bind_specs
+
             # Assign tests and run them
-            to_run = split_run(total_nodes, current_node)
-            parallel_run(to_run, current_node)
+            suite = split_run(total_nodes, current_node)
```
Comment (werkshy) [inline lib/kontrast/runner.rb]: nice renaming here

**Comment by werkshy** [comment]
This looks good, a nice change of terminology. :+1:

File: lib/kontrast/test_suite.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            @tests.each do |test|
+                suite_hash[test.width] ||= {}
+                suite_hash[test.width][test.name] = test.path
+            end
+
+            return suite_hash
+        end
+
+        private
+            def clear
```
Comment (irubnich) [inline lib/kontrast/test_suite.rb]: indent

File: lib/kontrast/spec_builder.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    end
+
+    class SpecBuilder
+        attr_reader :specs
+
+        def initialize
+            @specs = []
+        end
+
+        def add(spec_name)
```
Comment (irubnich) [inline lib/kontrast/spec_builder.rb]: add current_specs to specs

File: lib/kontrast/spec_builder.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                spec_folder = "./kontrast_specs"
+            end
+
+            spec_files = Dir[spec_folder + "/**/*_spec.rb"]
+            spec_files.each do |file|
+                require file
+            end
+        end
+
+        private
```
Comment (irubnich) [inline lib/kontrast/spec_builder.rb]: indent

Comment (irubnich) [inline lib/kontrast/spec_builder.rb]: remove private

Comment (irubnich) [inline lib/kontrast/spec_builder.rb]: clear!

File: lib/kontrast/runner.rb [hunk 6]
```diff
@@ -20,95 +17,89 @@ def run
...
+            parallel_run(suite, current_node)
         end
 
         # Given the total number of nodes and the index of the current node,
         # we determine which tests the current node will run
         def split_run(total_nodes, current_node)
-            all_tests = Kontrast.test_suite.tests
-            tests_to_run = Hash.new
+            test_suite = Kontrast.test_suite
+            tests_to_run = TestSuite.new
```
Comment (irubnich) [inline lib/kontrast/runner.rb]: Make this a regular list

**Pull request merged**